ui/src/components/layout/FlowCanvas.tsx

```diff
@@ -167,7 +173,7 @@ const FlowCanvas: React.FC<Props> = () => {
     }
   }
 
-  function handleAddNode(item: ThingInput) {
+  function handleAddNode(item: PathedData) {
```

Comment:
Since we don't need to serialize anything from the modal, we can just return to the whole object. This will let us fix the issue below pretty easily.